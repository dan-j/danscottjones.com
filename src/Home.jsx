import React from 'react';
import contentfulClient from './contentfulClient.js';


class Home extends React.Component {

    state = {
        entries: null,
    };

    componentDidMount() {
        contentfulClient.getEntries({ content_type: 'blogPost' })
            .then(entries => this.setState({ entries }));
    }

    render() {
        const entries = this.state.entries;
        let content;
        if (!entries) {
            content = <p>Loading...</p>
        } else {
            content = JSON.stringify(entries.items.map(item => item.fields));
        }

        return (
            <div>
                {content}
            </div>
        );
    }
}

export default Home;
