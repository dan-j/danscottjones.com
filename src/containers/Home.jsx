import React from 'react';
import contentfulClient from '../services/contentfulClient';
import BlogEntry from './BlogEntry';

class Home extends React.Component {

    state = {
        entries: null,
    };

    componentDidMount() {
        contentfulClient.getEntries({ content_type: 'blogPost' })
            .then(entries => this.setState({ entries }));
    }

    render() {
        const entries = this.state.entries;
        let content;
        if (!entries) {
            content = <p>Loading...</p>
        } else {
            content = entries.items.map(item => (
                <BlogEntry
                    key={item.sys.id}
                    title={item.fields.title}
                    body={item.fields.body}
                    slug={item.fields.slug}
                />
        ));
        }

        return (
            <div>
                {content}
            </div>
        );
    }
}

export default Home;
