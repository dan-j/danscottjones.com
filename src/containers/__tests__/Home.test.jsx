import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Home from '../Home';

jest.mock('../../services/contentfulClient', () => ({
    // eslint-disable-next-line global-require
    getEntries: () => require('./data/BlogPost').default.entries,
}));

test('<Home /> loads blogPost entries', (done) => {
    const component = renderer.create(
        <MemoryRouter>
            <Home />
        </MemoryRouter>,
    );

    expect(component.toJSON()).toMatchSnapshot();

    process.nextTick(() => {
        expect(component.toJSON()).toMatchSnapshot();
        done();
    });
});
