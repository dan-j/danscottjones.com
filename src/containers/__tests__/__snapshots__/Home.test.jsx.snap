// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> loads blogPost entries 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;

exports[`<Home /> loads blogPost entries 2`] = `
<div>
  <div
    className="blog-entry"
  >
    <a
      href="/blog/amazing-title"
      onClick={[Function]}
    >
      <h2>
        Amazing Title
      </h2>
      <p>
        What a body!
      </p>
    </a>
  </div>
</div>
`;
